crates/iota-transaction-builder/src/lib.rs

```diff
@@ -1,7 +1,277 @@
+//! offline client builder, but this will cause the builder to fail when calling
+//! `finish`.
+//!
+//! ## Methods
+//!
+//! There are three kinds of methods available:
+//!
+//! ### Commands
+//!
+//! Each command method adds one or more commands to the final transaction. Some
+//! commands have optional follow-up methods. All command results can be named
+//! via [name](TransactionBuilder::name);
+//!
+//! - [move_call](TransactionBuilder::move_call): Call a move function.
+//!     - `arguments`: Add arguments to the move call.
+//!     - `generics`: Add generic types to the move call using types that
+//!       implement [MoveType](types::MoveType).
+//!     - `type_tags`: Add generic types directly using the
+//!       [TypeTag](iota_types::TypeTag).
+//! - [send_iota](TransactionBuilder::send_iota): Send IOTA coins to a recipient
+//!   address.
+//! - [send_coins](TransactionBuilder::send_coins): Send coins of any type to a
+//!   recipient address.
+//! - [merge_coins](TransactionBuilder::merge_coins): Merge a list of coins into
+//!   a single primary coin.
+//! - [split_coins](TransactionBuilder::split_coins): Split a coin into coins of
+//!   various amounts.
+//! - [transfer_objects](TransactionBuilder::transfer_objects): Send objects to
+//!   a recipient address.
+//! - [publish](TransactionBuilder::publish): Publish a move package.
+//!     - `package_id`: Name the package ID returned by the publish call.
+//! - [upgrade](TransactionBuilder::upgrade): Upgrade a move package.
+//! - [make_move_vec](TransactionBuilder::make_move_vec): Create a move
+//!   `vector`.
+//!
+//! ### Metadata
+//!
+//! These methods set various metadata which may be needed for the execution.
+//!
+//! - [gas](TransactionBuilder::gas): Add a gas coin to pay for the execution.
+//! - [gas_coins](TransactionBuilder::gas_coins): Add gas coins to pay for the
+//!   execution.
+//! - [gas_budget](TransactionBuilder::gas_budget): Set the maximum gas budget
+//!   to spend.
+//! - [gas_price](TransactionBuilder::gas_price): Set the gas price.
+//! - [sponsor](TransactionBuilder::sponsor): Set the gas sponsor address.
+//! - [gas_station_sponsor](TransactionBuilder::gas_station_sponsor): Set the
+//!   gas station URL. See [Gas Station](crate#gas-station) for more info.
+//! - [expiration](TransactionBuilder::expiration): Set the transaction
+//!   expiration epoch.
+//!
+//! ### Other
+//!
+//! Many other methods exist, either to get data or allow for development on top
+//! of the builder. Typically, these methods should not be needed, but they are
+//! made available for special circumstances.
+//!
+//! - [apply_argument](TransactionBuilder::apply_argument)
+//! - [apply_arguments](TransactionBuilder::apply_arguments)
+//! - [input](TransactionBuilder::input)
+//! - [pure_bytes](TransactionBuilder::pure_bytes)
+//! - [pure](TransactionBuilder::pure)
+//! - [command](TransactionBuilder::command)
+//! - [named_command](TransactionBuilder::named_command)
```

Comment:
but shouldn't they be documented here as well?

Reply:
I think it's better to have the detailed docs on the method itself and just link to them here